src/Raven.Server/Documents/Indexes/CollectionOfBloomFilters.cs

```diff
@@ -95,14 +95,25 @@ internal unsafe BloomFilter CreateNewFilter(int number, Mode mode)
             Slice key;
             Slice.From(_context.Allocator, number.ToString("D9"), out key);
 
+            // we can safely pass raw pointers here and dispose DirectAdd scopes immediately because 
+            // filters' content will be written to overflows
+
             if (mode == Mode.X64)
             {
-                var ptr64 = _tree.DirectAdd(key, BloomFilter64.PtrSize);
-                return new BloomFilter64(key, ptr64, _tree, writeable: true);
+                Debug.Assert(_tree.ShouldGoToOverflowPage(BloomFilter64.PtrSize));
+
+                using (var ptr64 = _tree.DirectAdd(key, BloomFilter64.PtrSize))
+                {
+                    return new BloomFilter64(key, ptr64.Ptr, _tree, writeable: true);
+                }     
```

Comment:
See the comment on line 98, this is fine because they are overflow pages, so we know they can't be modified by page splits.